components/brave_federated/data_store_service.cc

```diff
@@ -5,28 +5,24 @@
 
 #include "brave/components/brave_federated/data_store_service.h"
 
+#include <memory>
 #include <utility>
 
 #include "base/memory/weak_ptr.h"
 #include "base/task/thread_pool.h"
 #include "base/threading/sequence_bound.h"
 #include "base/threading/sequenced_task_runner_handle.h"
-#include "brave/components/brave_federated/data_stores/ad_notification_timing_data_store.h"
+#include "brave/components/brave_federated/data_stores/data_store.h"
```

Comment:
already included in header. Please also make sure you forward declare in headers where possible